core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/aiservice/AiServiceMethodImplementationSupport.java

```diff
@@ -370,17 +371,10 @@ private List<ChatMessage> messagesToSend(UserMessage augmentedUserMessage,
             for (ToolExecutionRequest toolExecutionRequest : aiMessage.toolExecutionRequests()) {
                 log.debugv("Attempting to execute tool {0}", toolExecutionRequest);
                 ToolExecutor toolExecutor = toolExecutors.get(toolExecutionRequest.name());
-                if (toolExecutor == null) {
-                    throw runtime("Tool executor %s not found", toolExecutionRequest.name());
-                }
-                String toolExecutionResult = toolExecutor.execute(toolExecutionRequest, memoryId);
-                log.debugv("Result of {0} is '{1}'", toolExecutionRequest, toolExecutionResult);
-                ToolExecutionResultMessage toolExecutionResultMessage = ToolExecutionResultMessage.from(
-                        toolExecutionRequest,
-                        toolExecutionResult);
 
-                beanManager.getEvent().select(ToolExecutedEvent.class)
-                        .fire(new ToolExecutedEvent(auditSourceInfo, toolExecutionRequest, toolExecutionResult));
+                ToolExecutionResultMessage toolExecutionResultMessage = toolExecutor == null ?
+                        context.toolService.applyToolHallucinationStrategy(toolExecutionRequest) :
```

Comment:
This method has been introduced with [this commit](https://github.com/langchain4j/langchain4j/commit/171cd2918085a65df3307a4647feb73d274d04b7) so we need to wait to have it in a new langchain4j release before merging this pull request.